src/components/EmojiPicker/EmojiPicker.js

```diff
@@ -10,166 +10,143 @@ const DEFAULT_ANCHOR_ORIGIN = {
     vertical: CONST.MODAL.ANCHOR_ORIGIN_VERTICAL.BOTTOM,
 };
 
-class EmojiPicker extends React.Component {
-    constructor(props) {
-        super(props);
-
-        this.hideEmojiPicker = this.hideEmojiPicker.bind(this);
-        this.showEmojiPicker = this.showEmojiPicker.bind(this);
-        this.selectEmoji = this.selectEmoji.bind(this);
-        this.measureEmojiPopoverAnchorPosition = this.measureEmojiPopoverAnchorPosition.bind(this);
-        this.measureEmojiPopoverAnchorPositionAndUpdateState = this.measureEmojiPopoverAnchorPositionAndUpdateState.bind(this);
-        this.focusEmojiSearchInput = this.focusEmojiSearchInput.bind(this);
-        this.onModalHide = () => {};
-        this.onEmojiSelected = () => {};
-
-        this.state = {
-            isEmojiPickerVisible: false,
-
-            // The horizontal and vertical position (relative to the window) where the emoji popover will display.
-            emojiPopoverAnchorPosition: {
-                horizontal: 0,
-                vertical: 0,
-            },
-
-            emojiPopoverAnchorOrigin: DEFAULT_ANCHOR_ORIGIN,
-        };
+const EmojiPicker = forwardRef((props, ref) => {
+    const [isEmojiPickerVisible, setIsEmojiPickerVisible] = useState(false);
+    const [emojiPopoverAnchorPosition, setEmojiPopoverAnchorPosition] = useState({
+        horizontal: 0,
+        vertical: 0,
+    });
+    const [emojiPopoverAnchorOrigin, setEmojiPopoverAnchorOrigin] = useState(DEFAULT_ANCHOR_ORIGIN);
+    const [onModalHide, setOnModalHide] = useState(() => {});
+    const [onEmojiSelected, setOnEmojiSelected] = useState(() => {});
+    const [emojiPopoverAnchor, setEmojiPopoverAnchor] = useState(null);
+    const emojiSearchInput = useRef();
+
+    function measureEmojiPopoverAnchorPosition(anchorComponent) {
+        return new Promise((resolve) => {
+            if (!anchorComponent) {
+                return resolve({horizontal: 0, vertical: 0});
+            }
+            anchorComponent.measureInWindow((x, y, width) => resolve({horizontal: x + width, vertical: y}));
+        });
     }
 
-    componentDidMount() {
-        this.emojiPopoverDimensionListener = Dimensions.addEventListener('change', this.measureEmojiPopoverAnchorPositionAndUpdateState);
-    }
+    /**
+     * Show the emoji picker menu.
+     *
+     * @param {Function} [onModalHideValue=() => {}] - Run a callback when Modal hides.
+     * @param {Function} [onEmojiSelectedValue=() => {}] - Run a callback when Emoji selected.
+     * @param {Element} emojiPopoverAnchorValue - Element to which Popover is anchored
+     * @param {Object} [anchorOrigin=DEFAULT_ANCHOR_ORIGIN] - Anchor origin for Popover
+     * @param {Function} [onWillShow=() => {}] - Run a callback when Popover will show
+     */
+    function showEmojiPicker(onModalHideValue, onEmojiSelectedValue, emojiPopoverAnchorValue, anchorOrigin, onWillShow = () => {}) {
```

Comment:
why here creating the function with `function` and the other functions with `const` I would keep it consistent and this one use `const` too